subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -527,26 +541,9 @@ static void reg_step(struct k_work *work)
 	srv->reg.i = CLAMP(srv->reg.i, 0, UINT16_MAX);
 
 	float p = input * kp;
-	uint16_t output = CLAMP(srv->reg.i + p, 0, UINT16_MAX);
-
-	/* The regulator output is always in linear format. We'll convert to
-	 * the configured representation again before calling the Lightness
-	 * server.
-	 */
-	uint16_t lvl = light_get(srv);
-
-	/* Output value is max out of regulator and configured level. */
-	if (output > lvl) {
-		if (output == srv->reg.prev) {
-			return;
-		}
 
-		srv->reg.prev = output;
-		atomic_set_bit(&srv->flags, FLAG_REGULATOR);
-		light_set(srv, light_to_repr(output, LINEAR), REG_INT);
-	} else if (atomic_test_and_clear_bit(&srv->flags, FLAG_REGULATOR)) {
-		light_set(srv, light_to_repr(lvl, LINEAR), REG_INT);
-	}
+	srv->reg.out = CLAMP(srv->reg.i + p, 0, UINT16_MAX);
```

Comment:
Note that this is now causing floating point operations to be executed from interrupt context, where in the past they occurred in thread context.  Not sure whether that's an issue.